ggshield/sca/file_selection.py

```diff
@@ -0,0 +1,58 @@
+import re
+from pathlib import Path
+from typing import List, Set
+
+from ggshield.scan import Scannable
+from ggshield.scan.file import get_files_from_paths
+
+
+# List of filepaths to ignore for SCA scans
+
+SCA_IGNORE_LIST = (
+    "__pycache__",
+    ".git",
+    ".hg",
+    ".svn",
+    ".tox",
+    ".venv",
+    "site-packages",
+    ".idea",
+    "node_modules",
+    ".mypy_cache",
+    ".pytest_cache",
+    ".hypothesis",
+)
+
+
+def get_all_files_from_sca_paths(
+    path: Path,
+    exclusion_regexes: Set[re.Pattern],
+    verbose: bool,
+    ignore_git: bool = False,
+) -> List[str]:
+    """
+    Create a Files object from a path, recursively, ignoring non SCA files
+
+    :param path: path to scan
+    :param exclusion_regexes: list of regexes, used to exclude some filepaths
+    :param verbose: Option that displays filepaths as they are scanned
+    :param ignore_git: Ignore that the folder is a git repository. If False, only files tracked by git are scanned
+    """
+    files = get_files_from_paths(
+        paths=[str(path)],
+        exclusion_regexes=exclusion_regexes,
+        recursive=True,
+        yes=True,
+        verbose=verbose,
+        ignore_git=ignore_git,
+    ).apply_filter(is_not_excluded_from_sca)
+
+    return [str(x.relative_to(path)) for x in files.paths]
+
+
+def is_not_excluded_from_sca(scannable: Scannable) -> bool:
```

Comment:
Maybe `is_excluded_from_sca` which returns `any(part in SCA_IGNORE_LIST for part in scannable.path.parts)` would be more direct (and that would match the description in the docstring)

Reply:
It would be, however we rely on the `apply_filter` method of `Scannable` which requires to return `True` if the file is accepted. I modified the docstring, it was indeed outdated.

Reply:
Nit, but in this case I would find `return not any(... in SCA_IGNORE_LIST ...)` more direct (it factorizes the `not`)

Reply:
If you prefer ! Modified